internal/app/machined/internal/phase/rootfs/etc/etc.go

```diff
@@ -85,19 +86,27 @@ func Hosts(hostname string) (err error) {
 // ResolvConf copies the resolv.conf generated in the early boot to the new
 // root.
 func ResolvConf() (err error) {
-	target := "/run/system/etc/resolv.conf"
+	option := kernel.ProcCmdline().Get("ip").First()
+	switch option {
+	case nil:
+		target := "/run/system/etc/resolv.conf"
 
-	var f *os.File
+		var f *os.File
 
-	if f, err = os.OpenFile(target, os.O_WRONLY|os.O_CREATE, 0644); err != nil {
-		return err
-	}
+		if f, err = os.OpenFile(target, os.O_WRONLY|os.O_CREATE, 0644); err != nil {
+			return err
+		}
 
-	// nolint: errcheck
-	defer f.Close()
+		// nolint: errcheck
+		defer f.Close()
 
-	if err = unix.Mount("/run/system/etc/resolv.conf", "/etc/resolv.conf", "", unix.MS_BIND, ""); err != nil {
-		return fmt.Errorf("failed to create bind mount for /etc/resolv.conf: %w", err)
+		if err = unix.Mount("/run/system/etc/resolv.conf", "/etc/resolv.conf", "", unix.MS_BIND, ""); err != nil {
+			return fmt.Errorf("failed to create bind mount for /etc/resolv.conf: %w", err)
+		}
+	default:
+		if err = os.Symlink("/etc/resolv.conf", "/proc/net/pnp"); err != nil {
```

Comment:
This is probably fine, but I would actually check for the existence of `/proc/net/pnp` (and maybe even that it contains a line with "nameserver") instead of using the somewhat surrogate `ip` cmdline check.

Reply:
👍

Reply:
Actually, that file exists whether or not `ip=` was issued, so the existence check would be meaningless.